src/detectors/builtin/sendInLoop.ts

```diff
@@ -0,0 +1,340 @@
+import { CompilationUnit } from "../../internals/ir";
+import { foldStatements } from "../../internals/tact";
+import { MistiTactWarning, Severity } from "../../internals/warnings";
+import { ASTDetector } from "../detector";
+import {
+  AstStatement,
+  AstStatementExpression,
+  AstExpression,
+  AstCondition,
+  AstStatementTry,
+  AstStatementTryCatch,
+  AstStatementWhile,
+  AstStatementRepeat,
+  AstStatementUntil,
+  AstStatementForEach,
+  AstFunctionDef,
+  AstAsmFunctionDef,
+} from "@tact-lang/compiler/dist/grammar/ast";
+
+/**
+ * A detector that identifies send function calls inside loops and loops that may cause out-of-gas exceptions.
+ *
+ * ## Why is it bad?
+ * **Send Functions Inside Loops:**
+ * - **Unexpected Behavior:** Multiple sends may occur, leading to unintended contract interactions.
+ * - **High Gas Consumption:** Loops with send calls can consume excessive gas, potentially causing out-of-gas exceptions.
+ * - **Reentrancy Vulnerabilities:** Sending messages within loops might open up reentrancy attack vectors.
+ *
+ * **Loops with Excessive Iterations:**
+ * - **Out-of-Gas Exceptions:** Loops with a large number of iterations may consume all available gas.
+ * - **Denial of Service:** Excessive gas consumption can be exploited to create DoS attacks.
+ *
+ * ## Example
+ * ```tact
+ * // Send function inside a loop
+ * while (i < limit) {
+ *   send(SendParameters{ to: recipient, ... });
+ * }
+ *
+ * // Loop with excessive iterations
+ * repeat (1_000_001) {
+ *   // ...
+ * }
+ * ```
+ *
+ * **Recommendation:** Refactor the code to avoid calling send functions inside loops and reduce the number of loop iterations.
+ */
+export class SendInLoop extends ASTDetector {
+  severity = Severity.HIGH;
+
+  private functionDefinitions: Map<string, AstFunctionDef | AstAsmFunctionDef> = new Map();
```

Comment:
Does this detector support asm functions? If not, we could safely remove them from the map, right?

Reply:
How about `AstReceiver` and maybe `AstContractInit`?